src/ext/handlersToResp.c

```diff
@@ -194,11 +202,11 @@ static RdbRes toRespNewKey(RdbParser *p, void *userData, RdbBulk key, RdbKeyInfo
         return RDB_ERR_FAIL_ALLOC;
 
     /* apply del-key-before-write if configured. Not really required for strings */
-    if ((info->opcode != _RDB_TYPE_STRING) && (ctx->conf.delKeyBeforeWrite)) {
+    if ((info->opcode != _RDB_TYPE_STRING) && (ctx->delKeyBeforeWrite == DEL_KEY_BEFORE_BY_DEL_CMD)) {
```

Comment:
maybe improve the comment to mention why (i.e. that SET is an overwrite command)